### DISCUSSION TITLE
How are queries identified with keepPreviousData

### DISCUSSION THREAD

**Discussion by mwisnicki** [TanStack/query#4162] [Q&A] [answered]
As per https://tanstack.com/query/v4/docs/guides/paginated-queries when none sets `keepPreviousData`:

> The data from the last successful fetch available while new data is being requested, even though the query key has changed.

If the query key is ignored how can the QueryClient identify old data in the cache?

**Comment by TkDodo** [comment, marked as answer]
per observer. Every useQuery call creates an observer that knows it's previous state.
